src/mapping.rs

```diff
@@ -236,6 +283,53 @@ impl<'s> ProguardMapping<'s> {
     pub fn iter(&self) -> ProguardRecordIter<'s> {
         ProguardRecordIter { slice: self.source }
     }
+
+    /// Creates a [ClassIndex] from this mapping.
+    pub fn create_class_index(&self) -> ClassIndex {
+        let mut last_class = None;
+        let mut iter = self.iter();
+        let mut record_start = 0;
+        let mut index = BTreeMap::new();
+
+        while let Some(record) = iter.next() {
+            if let Ok(ProguardRecord::Class { obfuscated, .. }) = record {
+                if let Some((last_class_name, last_class_start)) = last_class.take() {
+                    index.insert(
+                        Cow::Borrowed(last_class_name),
+                        last_class_start..record_start,
+                    );
+                }
+                last_class = Some((obfuscated, record_start));
+            }
+            record_start = if iter.slice.is_empty() {
+                self.source.len()
+            } else {
+                std::cmp::min(
+                    iter.slice.as_ptr() as usize - self.source.as_ptr() as usize,
```

Comment:
maybe you can use [`byte_offset_from`](https://doc.rust-lang.org/std/primitive.pointer.html#method.byte_offset_from) for this? Casting to `usize` and doing math on it works in theory given that we know that `slice > source`, but using some safer variant might also be good.